compiler-plugin/src/test/kotlin/arrow/meta/quotes/scope/FunctionLiteralTest.kt

```diff
@@ -0,0 +1,55 @@
+package arrow.meta.quotes.scope
+
+import arrow.meta.plugin.testing.Code
+import arrow.meta.plugin.testing.CompilerTest
+import arrow.meta.plugin.testing.CompilerTest.Companion.source
+import arrow.meta.plugin.testing.assertThis
+import arrow.meta.quotes.scope.plugins.FunctionLiteralPlugin
+import io.kotlintest.specs.AnnotationSpec
+
+/**
+ * Function Literals are a little unclear - see more here: https://kotlinlang.org/docs/reference/lambdas.html
```

Comment:
I think you should just add here what you find confusing. In my case what I find most confusing is the many ways in which you can define function values but I understand there are ergonomic reasons and also it caters to a broader audience allowing more shapes and styles.

Reply:
I think I have some examples of clearer ideas I've added in this morning!